### PR TITLE
Add ARM64 kernels and images

### PR DESCRIPTION
**Pull request by luxas**
@stealthybox @chanwit Builds on top of #506, makes us have a prebuilt kernel and Ubuntu for ARM64

### CODE REVIEW

**Comment by stealthybox** [comment]
Related #345

**Comment by stealthybox** [comment]
Also see #513

**Comment by luxas** [comment]
This is ready to merge, ping @stealthybox

**Comment by darkowlzz** [comment]
Hi, I gave this a try on a [gentoo64-on-rpi-64bit](https://github.com/sakaki-/gentoo-on-rpi-64bit) machine and it worked well with docker and docker-bridge :slightly_smiling_face: 

With containerd and cni, I got the following error:
```
INFO[0005] Created VM with ID "d1b9bc96904c0061" and name "my-vm1" 
ERRO[0006] failed to setup network for namespace "ignite-d1b9bc96904c0061": failed to create bridge "ignite0": could not add "ignite0": operation not supported 
FATA[0006] failed to create bridge "ignite0": could not add "ignite0": operation not supported 
```
Need to debug and find out why it failed.

Used docker and docker-bridge and was able to create multiple VMs:
```
pi64 ~ # ignite run darkowlzz/ignite-ubuntu:18.04-v0.6.0-159-g858dad0-arm64 --kernel-image darkowlzz/ignite-kernel:4.19.97-arm64 --name my-vm1 --cpus 1 --ssh --runtime=docker --network-plugin=docker-bridge
INFO[0006] Created VM with ID "48197fcba7e146b9" and name "my-vm1" 
INFO[0012] Networking is handled by "docker-bridge"     
INFO[0012] Started Firecracker VM "48197fcba7e146b9" in a container with ID "720ce08ce0528c4ba8b31b29012a746a6fe996bb8b26a52f253493eaf15041e5" 

pi64 ~ # ignite run darkowlzz/ignite-ubuntu:18.04-v0.6.0-159-g858dad0-arm64 --kernel-image darkowlzz/ignite-kernel:4.19.97-arm64 --name my-vm2 --cpus 1 --ssh --runtime=docker --network-plugin=docker-bridge
INFO[0005] Created VM with ID "aabe60e7d1e6e6ed" and name "my-vm2" 
INFO[0007] Networking is handled by "docker-bridge"     
INFO[0007] Started Firecracker VM "aabe60e7d1e6e6ed" in a container with ID "c9af3ff989a485712fd7163b5f37965aae10ceefabb481f8be032a9b9fae2b02" 

pi64 ~ # ignite ps 
VM ID			IMAGE							KERNEL					SIZE	CPUS	MEMORY		CREATED		STATUS	IPS		PORTS	NAME
48197fcba7e146b9	darkowlzz/ignite-ubuntu:18.04-v0.6.0-159-g858dad0-arm64	darkowlzz/ignite-kernel:4.19.97-arm64	4.0 GB	1	512.0 MB	107s ago	Up 107s	172.17.0.2		my-vm1
aabe60e7d1e6e6ed	darkowlzz/ignite-ubuntu:18.04-v0.6.0-159-g858dad0-arm64	darkowlzz/ignite-kernel:4.19.97-arm64	4.0 GB	1	512.0 MB	7s ago		Up 7s	172.17.0.3		my-vm2
```
I was able to ssh into the VMs and use them.
```
pi64 ~ # ignite ssh my-vm1
...
root@localhost:~# uname -a
Linux localhost.localdomain 4.19.97 #1 SMP Sat Mar 14 21:47:48 UTC 2020 aarch64 aarch64 aarch64 GNU/Linux
```

Built all the images (kernel, OS and sandbox) for arm64 and ignite arm64.

This is great. Thanks!

File: images/Makefile [hunk 1]
```diff
@@ -35,8 +81,9 @@ build-all:
 	$(MAKE) ${OP} WHAT=alpine
 	$(MAKE) ${OP} WHAT=opensuse      RELEASE=leap       IS_LATEST=true
 	$(MAKE) ${OP} WHAT=opensuse      RELEASE=tumbleweed
-	$(MAKE) ${OP} WHAT=ubuntu        RELEASE=16.04
-	$(MAKE) ${OP} WHAT=ubuntu        RELEASE=18.04      IS_LATEST=true
-	$(MAKE) ${OP} WHAT=ubuntu        RELEASE=19.04
+	$(MAKE) ${OP} WHAT=ubuntu        RELEASE=16.04						IS_MANIFEST_LIST=0
+	$(MAKE) ${OP} WHAT=ubuntu        RELEASE=18.04      IS_LATEST=true	IS_MANIFEST_LIST=1 GOARCH=arm64
+	$(MAKE) ${OP} WHAT=ubuntu        RELEASE=18.04      IS_LATEST=true	IS_MANIFEST_LIST=1 GOARCH=amd64
```
Comment (stealthybox) [inline images/Makefile]: Ack -- only adding am arm64/aarch64 userspace image for Ubuntu 18.04 starting out

File: images/kernel/Dockerfile [hunk 2]
```diff
@@ -1,18 +1,37 @@
...
+# Fetch the latest tags from the linux kernel repo 
+RUN git fetch --tags
+
 ARG KERNEL_VERSION
 ARG KERNEL_EXTRA
-RUN git fetch --tags
+# ARCH here is KERNEL_ARCH in the Makefile. It needs to be hardcoded to ARCH for Kconfig to understand
+ARG ARCH
+ARG GOARCH
+ARG ARCH_MAKE_PARAMS
```
Comment (stealthybox) [inline images/kernel/Dockerfile]: Thanks for the comment

File: hack/push-manifest-list.sh [hunk 3]
```diff
@@ -0,0 +1,27 @@
+#!/bin/bash
+
+if [[ $# == 0 ]]; then
+    echo "Usage: $0 [manifest-list] [arch0...n]"
```
Comment (stealthybox) [inline hack/push-manifest-list.sh]: Nice move breaking this non-trivial manifest list logic into something that's generally useful for both the userspace and kernel images. :+1:

**Comment by luxas** [comment]
> @luxas did not report CNI issues in his experiments.

@darkowlzz @stealthybox This is why I upgraded CNI to v0.8.5: #561. https://github.com/containernetworking/plugins/pull/434

**Pull request merged**